test/unit/entity/user.test.js

```diff
@@ -300,4 +301,62 @@ describe('User', function() {
       expect(response).toEqual({});
     });
   });
+
+  describe('update()', function() {
+    it('should throw an error if the user does not have an _id', function() {
+      const user = new User({ email: randomString() });
+      return user.update({ email: randomString })
+        .catch((error) => {
+          expect(error).toBeA(KinveyError);
+          expect(error.message).toEqual('User must have an _id.');
+        });
+    });
+
+    it('should update the active user', function() {
+      return TestUser.logout()
+        .then(() => {
+          return TestUser.login(randomString(), randomString());
+        })
+        .then((user) => {
+          const email = randomString();
+          const requestData = assign(user.data, { email: email })
+          const responseData = assign(requestData, { _kmd: { authtoken: randomString() } });
+
+          // Kinvey API response
+          nock(this.client.apiHostname, { encodedQueryParams: true })
+            .put(`${user.pathname}/${user._id}`, requestData)
+            .reply(200, responseData);
+
+          return user.update({ email: email })
+            .then(() => {
+              const activeUser = User.getActiveUser();
+              expect(activeUser.data).toEqual(responseData);
+            });
+        });
+    });
+
+    it('should update a user and not the active user', function() {
+      return TestUser.logout()
+        .then(() => {
+          return TestUser.login(randomString(), randomString());
+        })
+        .then((activeUser) => {
+          const user = new User({ _id: randomString(), email: randomString() });
+          const email = randomString();
+          const requestData = assign(user.data, { email: email })
+          const responseData = assign(requestData, { _kmd: { authtoken: randomString() } });
+
+          // Kinvey API response
+          nock(this.client.apiHostname, { encodedQueryParams: true })
+            .put(`${user.pathname}/${user._id}`, requestData)
+            .reply(200, responseData);
+
+          return user.update({ email: email })
+            .then(() => {
+              expect(user.data).toEqual(responseData);
```

Comment:
This [line](https://github.com/Kinvey/js-sdk/pull/101/files#diff-ea2fab4905bfc00b94d59bb6949998e2R323) is where the `authtoken` is recreated. This [line](https://github.com/Kinvey/js-sdk/pull/101/files#diff-ea2fab4905bfc00b94d59bb6949998e2R333) verifies the active user contains the new data including the new `authtoken`.